pkg/apis/core/v1beta1/types_shoot.go

```diff
@@ -694,6 +694,12 @@ type KubeProxyConfig struct {
 	// defaults to IPTables.
 	// +optional
 	Mode *ProxyMode `json:"mode,omitempty" protobuf:"bytes,2,opt,name=mode,casttype=ProxyMode"`
+	// Enabled indicates whether kube-proxy should be deployed or not.
+	// Depending on your networking provider extensions switching kube-proxy off might be rejected.
+	// For example, cilium supports running without kube-proxy, but calico usually needs kube-proxy except for its ebpf dataplane.
+	// defaults to true if not specified.
+	// +optional
+	Enabled *bool `json:"enabled,omitempty" protobuf:"varint,3,opt,name=enabled"`
```

Comment:
Is it sensible/possible to change the value of this flag, i.e. to migrate from proxy enabled to proxy-less and vice-versa?
If not or if it is disruptive, I would vote for making the field immutable in validation for now.

Reply:
It is absolutely possible to switch between kube-proxy and kube-proxy-less mode. Depending on the networking extension, this might cause a short disruption as existing connections between pods and services might be reset. No user interaction required, though.
I would leave it as changeable. Any concerns?

Reply:
For posterity:
In an offline conversation @ScheererJ and I agreed on making the `kubeproxy.enabled` field immutable for now.
This is to support the transition phase, in which the current networking extension releases don't respect the field and would silently fail if kube-proxy is disabled but the dataplane doesn't support the proxy-less mode or is not configured to use it. By making the field immutable, this will prevent users from silently breaking service routing on existing clusters.

@ScheererJ argued that such misconfiguration can be easily fixed by flipping the switch again, but we agreed, that it isn't much effort and doesn't hurt to make the field immutable.
After a few gardener releases, networking extensions had enough time to support the flag and we can consider to make the field mutable again in gardener to enable the migration use case.

Reply:
Adapted the validation to make the field immutable for now in the newest commit.